components/sup/src/cli_test_helpers.rs

```diff
@@ -0,0 +1,151 @@
+use clap::{App, ArgMatches};
+use std::fmt;
+use std::iter::FromIterator;
+
+#[derive(Debug)]
+pub enum Expectation<'a> {
+    Value(&'a str),
+    Values(Vec<String>),
+}
+
+use self::Expectation::{Value, Values};
+
+#[derive(Debug)]
+enum CliTestError {
+    ValueMismatch(String, String, String),
+    MultiValueMismatch(String, Vec<String>, Vec<String>),
+    MissingFlag(String),
+    MissingSubcmd(),
+    ClapError(String),
+}
+
+impl fmt::Display for CliTestError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let content = match self {
+            CliTestError::ValueMismatch(flag, expected, actual) => format!(
+                r#"
+'{}' validation failed:
+  Expected: '{}'
+  Actual:   '{}'
+"#,
+                flag, expected, actual
+            ),
+            CliTestError::MultiValueMismatch(flag, expected, actual) => format!(
+                r#"
+'{}' validation failed:
+  Expected: '{:#?}'
+  Actual:   '{:#?}'
+"#,
+                flag, expected, actual,
+            ),
+            CliTestError::MissingFlag(flag) => format!("Expected {} flag was missing.", flag),
+            CliTestError::MissingSubcmd() => {
+                format!("Expected a subcommand, but none was specified")
+            }
+            CliTestError::ClapError(err) => format!(
+                r#"
+Command Parse error:
+****************************************
+{}
+****************************************
+"#,
+                err.to_string()
+            ),
+        };
+        write!(f, "{}", content)
+    }
+}
+
+pub fn assert_command(app: App, cmd: &str, assertions: Vec<(&str, Expectation)>) {
+    let cmd_vec = Vec::from_iter(cmd.split_whitespace());
+    let errors: Vec<CliTestError> = match app.get_matches_from_safe(cmd_vec) {
```

Comment:
Just a fun rust FYI, the compiler can infer the type here so 
```suggestion
    let errors = match app.get_matches_from_safe(cmd_vec) {
```
will work just as well.